--- virtual-machine.c-orig	2013-01-06 21:46:39.169082284 +0100
+++ virtual-machine.c	2013-01-06 21:46:27.043082554 +0100
@@ -1167,7 +1167,8 @@
   double f;
   int i;
   PARAMETER *opstack=*sp;
-  va_list arguments=*arg;
+  va_list arguments;
+  va_copy(arguments,*arg);
     switch(typ) {
     case 'a': a=va_arg ( arguments, ARRAY );  *((ARRAY *)(opstack->integer))=double_array(&a); opstack->typ=PL_ARRAY; opstack++; break;
     case 's': s=va_arg ( arguments, STRING ); *((STRING *)(opstack->integer))=double_string(&s); opstack->typ=PL_STRING; opstack++; break;
@@ -1180,7 +1181,7 @@
       break;
     }
     *sp=opstack;
-    *arg=arguments;
+    va_copy(*arg,arguments);
 }
 
 
